prometheus/pkg/reconciler/controller.go

```diff
@@ -58,7 +59,7 @@ func NewController(
 		prometheusClientSet:    client.Get(ctx),
 	}
 	impl := controller.NewImpl(r, r.Logger, ReconcilerName)
-
+	r.sinkResolver = resolver.NewURIResolver(ctx, impl.EnqueueKey)
 	r.sinkReconciler = duck.NewSinkReconciler(ctx, impl.EnqueueKey)
```

Comment:
We do not need this `sinkReconciler` anymore: `sinkResolver` should be the only thing you need